### PR TITLE
feat: add ETag support SDKCF-1088

### PR DESCRIPTION
**Pull request by donnie-jp**
# Description
- support ETag header and add tests
- fetch the config from a background queue
- create our own session instead of using shared session

## Links
SDKCF-1088

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I ran `fastlane ci` without errors

### CODE REVIEW

File: RRemoteConfig/Fetcher.swift [hunk 1]
```diff
@@ -13,14 +13,14 @@ internal class Fetcher {
...
-        request.addHeader("apiKey", "ras-\(environment.subscriptionKey)")
-        request.setExtraHeaders(from: environment)
+        request.setConfigHeaders(from: environment)
 
         apiClient.send(request: request, decodeAs: ConfigModel.self) { (result, response) in
             switch result {
             case .success(let resultConfig):
                 var config = resultConfig as? ConfigModel
                 config?.signature = response?.allHeaderFields["Signature"] as? String
+                self.environment.etag = response?.allHeaderFields["Etag"] as? String
```
Comment (corycaywood) [inline RRemoteConfig/Fetcher.swift:23]: What happens here on a `304` response? The `304` response doesn't have the ETag header, so will this still keep the ETag cached?

**Review by corycaywood — COMMENTED** [review]
Looks good, just one question.

**Review by corycaywood — APPROVED** [review]

File: RRemoteConfig/Fetcher.swift (hunk 1 above)
Comment (donnie-jp) [reply RRemoteConfig/Fetcher.swift:23]: I actually get a http status `200` response from the URLSession's url cache.
For the same request/response Charles shows the expected `304`.
Presumably URLSession's cache sends back the entire previous response.

Comment (corycaywood) [reply RRemoteConfig/Fetcher.swift:23]: Ah, okay that makes sense.

**Pull request merged**